### PR TITLE
Bug fix for Issue with caseworkers failing when logged in as case officer

### PR DESCRIPTION
**Pull request by udaydenduluri33**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/EUI-3602

### Change description ###

Issue with caseworkers failing when logged in as case officer

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: api/workAllocation/index.ts [hunk 1]
```diff
@@ -63,8 +63,7 @@ export async function searchTask(req: EnhancedRequest, res: Response, next: Next
     res.status(status);
     // Assign actions to the tasks on the data from the API.
     if (data) {
-      const caseworkers: Caseworker[] = await retrieveAllCaseWorkers(req, res);
-      assignActionsToTasks(data.tasks, req.body.view, caseworkers);
+      assignActionsToTasks(data.tasks, req.body.view, req.session.caseworkers);
```
Comment (paul-graham) [inline api/workAllocation/index.ts:66]: I don't understand why this change has been made. The `await retrieveAllCaseWorkers(req, res)` call was already returning it from `req.session.caseworkers` if it was on there and, if not, it was loading the caseworkers.

**Review by paul-graham — CHANGES_REQUESTED** [review]

File: api/workAllocation/index.ts (hunk 1 above)
Comment (udaydenduluri33) [reply api/workAllocation/index.ts:66]: This is made because the normal caseworker who is logged in cannot access caseworkers api. Its easier to show rather than type

Comment (udaydenduluri33) [reply api/workAllocation/index.ts:66]: If I am logged in as case worker the assignee name is not required.
Only if I am logged in as senior case worker the assignee name is required

**Review by adnanakgun — APPROVED** [review]

**Review by ronaldmansveld — APPROVED** [review]

**Review by paul-graham — CHANGES_REQUESTED** [review]
It has been explained to me by @udaydenduluri33 (on Slack) that the change in `searchTask(...)` is to resolve an issue where the call to retrieve caseworkers will fail for users that don't have permission to retrieve them.

My suggestion remains that the better solution is either:
1. Identify prior to making the call the roles that have permission to retrieve them and only make the call for those roles, or;
2. Handle the first failure gracefully and then make no subsequent calls for the logged in user with the knowledge that they have no permission.

What is _not_ a good idea is to simply bypass making the call and introduce a race condition where you're relying on another call having been made previously. If you have call chaining in place to ensure the first call completes before this call could ever be made, pass the list of already-retrieved caseworkers down to this method rather than relying on an indirect method for access.

Neither manual testing nor automation testing will reliably catch a race condition - the solution _must_ be within the code.

**Comment by udaydenduluri33** [comment]
> * Identify prior to making the call the roles that have permission to retrieve them and only make the call for those roles, or;
> * Handle the first failure gracefully and then make no subsequent calls for the logged in user with the knowledge that they have no permission.

@paul-graham 
Tried to answer that this will not have any race conditional issues. I can explain in detail if you want in a call what this fix is all about.

File: api/workAllocation/index.ts (hunk 1 above)
Comment (paul-graham) [reply api/workAllocation/index.ts:66]: Where is that logic in the code?
